_unittest_solvers/test_26_emit.py

```diff
@@ -1170,3 +1171,92 @@ def test_result_categories(self, add_app):
+            best_freq = None
+
+            combinations_run = 0
+            
+            domain.set_receiver(receiver)
+            interaction = rev.run(domain)
+
+            for interferer in interferers:
+                for tx_band in rev.get_band_names(interferer, modeTx):
+                    for tx_freq in rev.get_active_frequencies(interferer, tx_band, modeTx):
+                        domain.set_interferer(interferer, tx_band, tx_freq)
+                        rx_band = rev.get_band_names(receiver, modeRx)[0]
+                        for rx_freq in rev.get_active_frequencies(receiver, rx_band, modeRx):
+                            
+                            domain.set_receiver(receiver, rx_band, rx_freq)
+
+                            if best_band == None:
+                                best_band = rx_band
+                            if best_freq == None:
+                                best_freq = rx_freq
+
+                            try: 
+                                instance = interaction.get_instance(domain)
+                                if instance.has_valid_values():
+                                    emi = instance.get_value(modeEmi)
+                                    if emi < best_emi:
+                                        best_emi = emi
+                                        best_band = rx_band
+                                        best_freq = rx_freq
+                                else:
+                                    assert(f'No valid values found')
+                            except:
+                                assert("No results between " + interferer + ": " + tx_band + 
+                                       ": " + str(tx_freq) + " and " + receiver + ": " + 
+                                       rx_band + ": " + str(rx_freq))
+                            
+                            combinations_run += 1
+
+                            if combinations_run >= 20:
+                                return [best_band, best_freq]
+            return [best_band, best_freq]
+        
+        # Warmup in case something isn't solved
+        best_case_rx_ch = {}
+        for rx in [receivers[0]]:
+            best_case_rx_ch[rx] = get_best_rx_channel(results, rx)
+        
+        # Get the time without the license session
+        start = time.perf_counter()
+        best_case_rx_ch = {}
+        for rx in [receivers[0]]:
+            best_case_rx_ch[rx] = get_best_rx_channel(results, rx)
+        end = time.perf_counter()
+        noLicenseSessionTime = end - start
+        
+        # Get the time using the license session
+        start = time.perf_counter()
+        best_case_rx_ch = {}
+        with revision.get_license_session():
+            for rx in [receivers[0]]:
+                best_case_rx_ch[rx] = get_best_rx_channel(results, rx)
+        end = time.perf_counter()
+        licenseSessionTime = end - start
+
+        assert licenseSessionTime < noLicenseSessionTime
```

Comment:
@bryankaylor Should we make sure it's at least 2x faster? A simple `<` could pass due to the times being just slightly different (noise essentially). 

Also, if the results are cached by the first loop, we might not be solving in the second loop which could also cause it to be faster without really verifying the license session is helping. I'm not sure this could happen right now, but a change to caching could mean it starts happening. Ideally we'd ensure nothing is cached/solved the second time through by purging results. I don't think there is a way to do that right now, but would reversing the order help? If we run the license session first which should be faster, things would fail if the second uses a cache and is faster because it doesn't actually have to run and thus doesn't do license checkout. Does that make sense?

Reply:
@myoung301 - There is an untimed warmup loop on line 1240, before the two timed loops. I put it there in case there is or will be some caching behavior that affects the times. With it, either the results will be cached for both timed loops or for neither. Does that sound right?

I do agree we should change it to licenseSessionTime*2 < noLicenseSessionTime, so if this caching starts happening and the times are nearly equal (or if the times are nearly equal for any other reason), it will fail consistently instead of being almost random.